src/sentry/api/event_search.py

```diff
@@ -142,6 +142,16 @@ def __str__(self):
             map(six.text_type, (self.key.name, self.operator, self.value.raw_value)),
         )
 
+    @cached_property
+    def is_negation(self):
+        # Negations are mostly just using != operators. But we also have
+        # negations on has: filters, which translate to = '', so handle that
+        # case as well.
+        return (
+            self.operator == '!=' and self.value.raw_value != ''
+            or self.operator == '=' and self.value.raw_value == ''
```

Comment:
mm i think i'm not fully understanding the `self.operator == '=' and self.value.raw_value == ''` case, but could this also return true for something like `tag:""`?

Reply:
Yeah, although I think a tag being equal to an empty string is equivalent to it being null?

`!has:abc` and `abc:""` both produce `key=SearchKey(name='abc'), operator='=', value=SearchValue(raw_value=''))`

I guess it might be confusing for users if they get blocked from that kind of query, but I don't know that it's all that valid? 

Another option could be to store whether a filter is a negation just as a value on the `SearchFilter`, then we can just use that directly and avoid this logic. If you think it's worthwhile I'll follow up with that change.

Reply:
ah i see. ok i think let's just leave as is and then if people are confused, we can change